src/vm-repair/azext_vm_repair/custom.py

```diff
@@ -0,0 +1,215 @@
+                                            .format(g=resource_group_name, n=storage_account.account_name)
+            logger.info('Fetching storage account connection string:')
+            connection_string = _call_az_command(get_connection_string_command).strip('\n')
+
+            # Create Snapshot of Unmanaged Disk
+            make_snapshot_command = 'az storage blob snapshot -c {c} -n {n} --connection-string "{con_string}" --query snapshot -o tsv' \
+                                    .format(c=storage_account.container, n=storage_account.blob, con_string=connection_string)
+            logger.info('Creating snapshot of OS disk:')
+            snapshot_timestamp = _call_az_command(make_snapshot_command).strip('\n')
+            snapshot_uri = os_disk_uri + '?snapshot={timestamp}'.format(timestamp=snapshot_timestamp)
+
+            # Copy Snapshot into unmanaged Disk
+            copy_snapshot_command = 'az storage blob copy start -c {c} -b {name} --source-uri {source} --connection-string "{con_string}"' \
+                                    .format(c=storage_account.container, name=copied_os_disk_name, source=snapshot_uri, con_string=connection_string)
+            logger.info('Creating a copy disk from the snapshot:')
+            _call_az_command(copy_snapshot_command)
+             # Generate the copied disk uri
+            copied_disk_uri = os_disk_uri.rstrip(storage_account.blob) + copied_os_disk_name
+
+            # Create new rescue VM with copied ummanaged disk command
+            create_rescue_vm_command = create_rescue_vm_command + ' --use-unmanaged-disk'
+            logger.info('Creating rescue vm while disk copy is in progress:')
+            _call_az_command(create_rescue_vm_command)
+
+            logger.info('Checking if disk copy is done:')
+            copy_check_command = 'az storage blob show -c {c} -n {name} --connection-string "{con_string}" --query properties.copy.status -o tsv' \
+                                 .format(c=storage_account.container, name=copied_os_disk_name, con_string=connection_string)
+            copy_result = _call_az_command(copy_check_command).strip('\n')
+            if copy_result != 'success':
+                # TODO, what then if unsucessful. Call again or exit?
+                # Let it run for now and see how the disk state it.
+                logger.warning('Disk copy UNSUCCESSFUL.')
+
+            # Attach copied unmanaged disk to new vm
+            logger.info('Attaching copied disk to rescue VM as data disk:')
+            attach_disk_command = "az vm unmanaged-disk attach -g {g} -n {disk_name} --vm-name {vm_name} --vhd-uri {uri}" \
+                                  .format(g=rescue_rg_name, disk_name=copied_os_disk_name, vm_name=rescue_vm_name, uri=copied_disk_uri)
+            _call_az_command(attach_disk_command)
+        
+    # Some error happened. Stop command and clean-up resources.
+    except Exception as exception:
```

Comment:
Why is this exception broad? Its okay if its not feasible to make this more specific, given all what is going on here. But if you can make this more specific, that would be a plus. 

At least you can try to handle KeyboardInterrupt and / or SystemExit (simply reraise it) separately